---
title: "Lymphangioma Sequencing"
author: "Brian Mannakee"
date: "12/14/2017"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
source('./analyze_mutect2_calls.R')
```

## Summary

The Whole Genome Sequence of this patient's tumor is of high quality, but due to low tumor cellularity is not sufficient to confidently identify all clonal mutations that are present in the tumor.
The evidence for this is that the PIK3CA H1047R mutation identified by Foundation One is not called by any of the three variant callers we tried.
We can, however, make a couple of assumptions that make the data useful.
The PIK3CA mutation is *seen* by the variant callers, but the *evidence* from the alternate reads at this site is just below the threshold needed to confidently call the site variant.
By adjusting the evidence threshold down to meet the standard provided by the PIK3CA mutation, it is possible to generate a relatively high-confidence set of variant calls.
Alone this would not be enough to declare the variants described below as the *complete* set of variants present in the tumor, but combined with other information it may be sufficient for the study at hand.
The sequencing depth of this sample is not sufficient to make determinations (or even guesses) about copy number variation.


## Sequencing characteristics

The sequencing was carried out to achieve roughly 30X coverage.
Sequencing quality is good, for both the tumor and normal samples.
Based on allele frequencies the sample appears to be roughly 25\% tumor and 75\% normal, which is in line with expectations for this type of tumor.
The large number of variants with allele frequencies at 10-12\% likely represent clonal or nearly clonal heterozygous variants.

```{r vaf_histogram, echo=FALSE,warning=FALSE,fig.width=6,fig.height=4}
pl1
```

\newpage

## Variant calling

Tumor variants were called with MuTect2, MuTect1, and Varscan with very similar results, and the MuTect2 mutations are presented here.
At 30X depth, a clonal heterozygous variant in this sample is only likely to have 3-4 alternate (non-reference) reads present, which presents a challenge for variant calling, because the evidence used to classify a site as variant is proportional to the number of alternate reads observed.
For areas of moderately low coverage only 2 reads are sufficient for calling a clonal heterozygous variant, but it is not generally advisable to call variants with alternate read counts that low since the possibility that these are errors is quite high.
The coding variants that pass all MuTect2 quality filters and have 3 or more alternate reads are in the table below.



```{r passed_table,echo=FALSE}
library(knitr)


tbl1 <- filter_fr %>% dplyr::filter(all_filters=="PASS" & alt_depth > 2) %>%
  rename("Gene_Name"=Hugo_Symbol,
         "VAF"=tumor_f,
         "Evidence_level"=t_lod_fstar__INFO__) %>%
  replace_na(list(cgc_somatic="no",
                  cgc_syndrome="",
                  mutsig_results="",
                  cosmic_overlapping="")) %>%
  dplyr::select(Gene_Name,Protein_Change,VAF,alt_depth,ref_depth,"CGC Syndrome"=cgc_syndrome,"COSMIC"=cosmic_overlapping)
kable(tbl1)
```

Restricting the callset to those variants with more than two alternate reads excludes 10 called variants, but only one of these is annotated as significant by the Cancer Genome Consortium(CGC), COSMIC, or MUTSIG.
The significant variant is a p.Y2162fs frame shift deletion in NF1, covered by 2 alternate reads and 19 reference reads, which was not observed in the targeted deep sequencing done by Foundation One.
Among the called coding variants only the NAT2 variant p.R268K is annotated in COSMIC, but it is not annotated as significant by CGC or MUTSIG.

\newpage
Unfortunately, we know from Foundation One sequencing that there is a significant PIK3CA H1047R mutation, and it is also not called.
There are three alternate reads at this position in the tumor, but they have slightly below average base quality scores which causes the evidence for a variant to fall below the threshold that MuTect uses to classify sites as variant.
No combination of variant callers and parameter adjustments we have found will cause a variant caller to identify this site as variant.
However, since we have independent evidence that this site is truly variant we can adjust the evidence threshold.
The table below shows all sites with three or more variant reads, passing all MuTect filters except the evidence filter, and further restricting the results to those variants that are annotated by one or more of CGC, COSMIC (suppressed in the table below for space), or MUTSIG.

```{r final_table,echo=FALSE}
tbl2 <- final_fr %>% dplyr::filter(alt_depth > 2) %>%
  dplyr::select(Gene_Name,Protein_Change,VAF,"CGC Syndrome"=cgc_syndrome,"MUTSIG"=mutsig_results)
kable(tbl2)
```


